config/routes.rb

```diff
@@ -1,6 +1,18 @@
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
+  mount Rswag::Ui::Engine => '/api-docs'
+  mount Rswag::Api::Engine => '/api-docs'
+  
+  namespace :api do
+    namespace :v1 do
+      post "/login", to:"users#login"
+      post "/register", to:"users#register"
 
-  # Defines the root path route ("/")
-  # root "articles#index"
+      get "users/:id/reservations", to:"users#reservations",as: "reservations"
+
+      post "/reservations/user/:id", to: "reservations#reserve"
+      delete "/reservations/user/:id", to: "reservations#checkout"
+      resources :motorcycles
+      resources :users
```

Comment:
- [ ] OPTIONAL: Since you are not using create, show, edit, update, or destroy in the users controller, you don't need `resources :users`.  It's just creating routes that are not being used.

Since I suspected this from the beginning, I made the whole review after removing this line and it all worked wonderfully with no issues (tested login, logout, re-login, reserving, and accessing my reservations).

If you are not using create, show edit, update, or destroy with users, kindly remove this `resources :users` line.   If for some reason I couldn't figure out you still need it, by all means leave it there 👍